otherlibs/dune-rpc/private/exported_types.ml

```diff
@@ -34,6 +34,468 @@ module Loc = struct
   ;;
 end
 
+let sexp_pp (conv_tag : 'a Conv.value) : 'a Pp.t Conv.value =
+  let open Conv in
+  let open Pp.Ast in
+  let nop = constr "Nop" unit (fun () -> Nop) in
+  let verbatim = constr "Verbatim" string (fun s -> Verbatim s) in
+  let char = constr "Char" char (fun c -> Char c) in
+  let newline = constr "Newline" unit (fun () -> Newline) in
+  let t =
+    fixpoint (fun t ->
+      let text = constr "Text" string (fun s -> Text s) in
+      let seq = constr "Seq" (pair t t) (fun (x, y) -> Seq (x, y)) in
+      let concat = constr "Concat" (pair t (list t)) (fun (x, y) -> Concat (x, y)) in
+      let box = constr "Box" (pair int t) (fun (x, y) -> Box (x, y)) in
+      let vbox = constr "Vbox" (pair int t) (fun (x, y) -> Vbox (x, y)) in
+      let hbox = constr "Hbox" t (fun t -> Hbox t) in
+      let hvbox = constr "Hvbox" (pair int t) (fun (x, y) -> Hvbox (x, y)) in
+      let hovbox = constr "Hovbox" (pair int t) (fun (x, y) -> Hovbox (x, y)) in
+      let break =
+        constr
+          "Break"
+          (pair (triple string int string) (triple string int string))
+          (fun (x, y) -> Break (x, y))
+      in
+      let tag = constr "Tag" (pair conv_tag t) (fun (s, t) -> Tag (s, t)) in
+      sum
+        [ econstr nop
+        ; econstr verbatim
+        ; econstr char
+        ; econstr newline
+        ; econstr text
+        ; econstr seq
+        ; econstr concat
+        ; econstr box
+        ; econstr vbox
+        ; econstr hbox
+        ; econstr hvbox
+        ; econstr hovbox
+        ; econstr break
+        ; econstr tag
+        ]
+        (function
+         | Nop -> case () nop
+         | Seq (x, y) -> case (x, y) seq
+         | Concat (x, y) -> case (x, y) concat
+         | Box (i, t) -> case (i, t) box
+         | Vbox (i, t) -> case (i, t) vbox
+         | Hbox t -> case t hbox
+         | Hvbox (i, t) -> case (i, t) hvbox
+         | Hovbox (i, t) -> case (i, t) hovbox
+         | Verbatim s -> case s verbatim
+         | Char c -> case c char
+         | Break (x, y) -> case (x, y) break
+         | Newline -> case () newline
+         | Text s -> case s text
+         | Tag (s, t) -> case (s, t) tag))
+  in
+  let to_ast x =
+    match Pp.to_ast x with
+    | Ok s -> s
+    | Error () ->
+      (* We don't use the format constructor in dune. *)
+      assert false
+  in
+  iso t Pp.of_ast to_ast
+;;
+
+module Ansi_color = struct
+  module RGB8 = struct
+    type t = char
+
+    let sexp = Conv.char
+    let to_ansi_color_rgb8 t = Stdune.Ansi_color.RGB8.of_char t
+    let of_ansi_color_rgb8 t = Stdune.Ansi_color.RGB8.to_char t
+  end
+
+  module RGB24 = struct
+    type t = int
+
+    let sexp = Conv.int
+    let to_ansi_color_rgb24 t = Stdune.Ansi_color.RGB24.of_int t
+    let of_ansi_color_rgb24 t = Stdune.Ansi_color.RGB24.to_int t
+  end
+
+  module Style = struct
+    type t =
```

Comment:
Ansi_color.Style in Stdune uses polymorphic variants for this type as there is a need to add constructors for parsing reasons. I originally tried to do the same here thinking that it would allow us to avoid the conversion functions however the compiler didn't allow it. Therefore for clarity I made this a regular variant with explicit conversion functions.

Reply:
Fixed this.